Runtime/Scripts/DataCollection/RGDataCollection.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using RegressionGames.StateActionTypes;
+using RegressionGames.Types;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace RegressionGames.DataCollection
+{
+    
+    /**
+     * An RGDataCollection instance is responsible for coordinating all data collection features on behalf
+     * of Regression Games. It will collect the replay data, logs, screenshots, and then upload these items
+     * to Regression Games. Currently only supports screenshots, with more to come in an upcoming PR.
+     */
+    public class RGDataCollection
+    {
+
+        private readonly string _sessionName;
+        private GameObject _parent;
+
+        public RGDataCollection(GameObject parent)
+        {
+            // Name the session, and setup a temporary directory for all data
+            _sessionName = Guid.NewGuid().ToString();
+            _parent = parent;
+        }
+
+        public void CaptureScreenshot(long tick)
+        {
+            Debug.Log($"Captured screenshot at tick {tick}");
+            string path = GetSessionDirectory($"screenshots/{tick}.jpg");
+
+            var texture = ScreenCapture.CaptureScreenshotAsTexture(1);
+
+            // Encode the texture into a jpg byte array
+            byte[] bytes = texture.EncodeToJPG(100);
+
+            // Save the byte array as a jpg file
+            File.WriteAllBytes(path, bytes);
+
+            // Destroy the texture to free up memory
+            Object.Destroy(texture);
+        }
+
+        public void RecordSession(long botInstanceId, RGClientConnectionType rgClientConnectionType)
+        {
+            Debug.Log("Ending data collection, uploading data to Regression Games...");
+
+            // First, upload all of the screenshots
+            var screenshotFiles = Directory.GetFiles(GetSessionDirectory($"screenshots"));
+            foreach (var screenshotFilePath in screenshotFiles)
+            {
+                var screenshotFile = Path.GetFileName(screenshotFilePath);
+                var tick = long.Parse(screenshotFile.Split(".")[0]);
+                RGServiceManager.GetInstance()?.UploadScreenshot(
+                    botInstanceId, tick, screenshotFilePath,
+                    () =>
+                    {
+                        Debug.Log($"Successfully uploaded screenshot for bot instance {botInstanceId} and tick {tick}");
+                    },
+                    () =>
+                    {
+                        
+                    });
+            }
+
+            Debug.Log("Data uploaded to Regression Games");
+            
+        }
+
+        private string GetSessionDirectory(string path = "")
+        {
+            var fullPath = Path.Join(Application.persistentDataPath, $"RGData/{_sessionName}", path);
```

Comment:
should separate `$"RGData/{_sessionName}"` into `"RGData", _sessionName` so that the path helper is building it OS agnostic

Reply:
Good catch, missed that one 🤣

Reply:
The reason I did this was because Join only supports up to 3 path pieces (very strange) - updated this to use Combine which does supports a variable number of path pieces, and works as long as the first path is an absolute path (which it is in this case).